Feature/singleton

xfn3t opened the pull request.

xfn3t committed:
add singleton realization

xfn3t committed:
add base tests

xfn3t committed:
git env

xfn3t committed:
Merge branch 'main' of https://github.com/xfn3t/webbee-homework into feature/singleton

gobaz095 reviewed (changes requested):
Те же замечания что и к https://github.com/xfn3t/webbee-homework/pull/8#pullrequestreview-2969460016

xfn3t committed:
refactor!: migrate project to Maven structure

Changes:
- Added pom.xml with project configuration
- Updated Main.java to work with Maven
- Modified Singleton.java for Maven compatibility
- Created SingletonTest.java in test directory

Project structure changed:
  src/main/java/ - main source code
  src/test/java/ - test source code

BREAKING CHANGE: Project now requires Maven for building.
All dependencies must be declared in pom.xml.
Run 'mvn clean install' to build the project.
